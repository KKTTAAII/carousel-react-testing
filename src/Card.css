.Card {
  display: flex;
  flex-direction: column;
}

.Card-title {
  margin: 0 0 0.5em 0;
  font-weight: bold;
}

.Card-small {
  margin-top: 0.5em;
  font-style: italic;
  font-size: small;
}

.Card-image {
  border-radius: 8px;
}