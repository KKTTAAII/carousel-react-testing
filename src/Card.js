import React from "react";
import "./Card.css";

const Card = ({caption, src, totalNum, currNum}) => {
  return (
    <div className="Card">
      <span className="Card-title">{caption}</span>
      <img className="Card-image" src={src} alt={caption} />
      <span className="Card-small">
        Image {currNum} of {totalNum}.
      </span>
    </div>
  );
}

export default Card;
