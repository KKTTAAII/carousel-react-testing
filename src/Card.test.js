import React from "react";
import { render, fireEvent } from "@testing-library/react";
import Card from "./Card";

it("renders Card component successfully", () => {
    render(<Card />)
});

it("matches snapshot", () => {
    const {asFragment} = render(<Card />);
    expect(asFragment()).toMatchSnapshot();
});