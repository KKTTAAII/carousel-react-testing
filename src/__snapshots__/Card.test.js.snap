// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="Card"
  >
    <span
      class="Card-title"
    />
    <img
      class="Card-image"
    />
    <span
      class="Card-small"
    >
      Image  of .
    </span>
  </div>
</DocumentFragment>
`;
